### ISSUE TITLE
Align SELECT...FROM...JOIN 

### ISSUE THREAD

**Issue by JoachimEck** [SAP/abap-cleaner#332]
Hey,

I think something is not rigth with the "Align SELECT ... FROM ... JOIN" rule. I was wondering why the Cleaner kept messing up the formatting in a SELECT. 
I'm using version 1.16.2.

1. align AS always seems to be done, regardless of the checkbox ![image](https://github.com/SAP/abap-cleaner/assets/101569831/9c13a229-e3cc-42f4-ac75-c18b53ca58d7) ![image](https://github.com/SAP/abap-cleaner/assets/101569831/9188d71f-2fc4-4d6d-983d-80c69e6f3d92)

2. if I turn on the checkbox for the "Align ON", then it moves the ON behind the upper ON condition. If the checkbox is off, then it does what I would have expected if the checkbox was on ![image](https://github.com/SAP/abap-cleaner/assets/101569831/a8992a50-bdc6-4ba2-9fdf-1f88f11afcf5) ![image](https://github.com/SAP/abap-cleaner/assets/101569831/5b5229e9-a7ec-4695-a212-5e57f7f64f56)

Maybe I just misinterpreted the options and everything is as it should be 😄

```ABAP   
 SELECT *
      FROM test
           JOIN join_one ON test~aufnr = jo~aufnr
           JOIN join_two 		
		    			ON  test~aufnr = jt~aufnr
										AND test~werks = jt~werks
      INTO CORRESPONDING FIELDS OF TABLE @lt_table. 
```
Here is my profile: [profile.zip](https://github.com/user-attachments/files/15983533/profile.zip)

Best regards,
Joachim

**Comment by jmgrassau** [comment]
Hi Joachim, 

thanks a lot for reporting this – ABAP cleaner didn't correctly determine the end of first the logical expressions here (ON …), which only happened if it was followed by JOIN (rather than INNER JOIN, LEFT|RIGHT OUTER JOIN, CROSS JOIN). This will be fixed with the next release, and with that, the options should work as expected. 

![image](https://github.com/SAP/abap-cleaner/assets/79639298/0614eb94-d81c-4aab-9639-9cad7733e71c)

![image](https://github.com/SAP/abap-cleaner/assets/79639298/0fd3fd4e-73d8-4265-a5b3-a441419a4718)

Kind regards,
Jörg-Michael

**Comment by jmgrassau** [comment]
Hi Joachim, 

thanks again for reporting this issue – this should now be fixed with version 1.18.0, which was just released!

Kind regards,
Jörg-Michael
